src/agent0/chainsync/db/hyperdrive/convert_data.py

```diff
@@ -6,107 +6,392 @@
+        )
+        as_base_idx = events_df["asBase"] & events_idx
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    # Shorts
+    events_idx = events_df["event"].isin(["OpenShort", "CloseShort"])
+    if events_idx.any():
+        events_df.loc[events_idx, "token_type"] = "SHORT"
+        events_df.loc[events_idx, "token_id"] = "SHORT-" + events_df.loc[events_idx, "maturityTime"].astype(int).astype(
+            str
+        )
+
+    events_idx = events_df["event"] == "OpenShort"
+    if events_idx.any():
+        # Pandas apply doesn't play nice with types
+        events_df.loc[events_idx, "token_delta"] = events_df.loc[events_idx, "bondAmount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        as_base_idx = events_df["asBase"] & events_idx
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = -events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = -events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    events_idx = events_df["event"] == "CloseShort"
+    if events_idx.any():
+        # Pandas apply doesn't play nice with types
+        events_df.loc[events_idx, "token_delta"] = -events_df.loc[events_idx, "bondAmount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        as_base_idx = events_df["asBase"] & events_idx
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    # Add solo transfer events to events_df
+    events_df = pd.concat([events_df, transfer_events_df], axis=0)
+
+    # We select the subset of columns we need and rename to match db schema
+    rename_dict = {
+        "address": "hyperdrive_address",
+        "transactionHash": "transaction_hash",
+        "blockNumber": "block_number",
+        "trader": "wallet_address",
+        "event": "event_type",
+        "token_type": "token_type",
+        "maturityTime": "maturity_time",
+        "token_id": "token_id",
+        "token_delta": "token_delta",
```

Comment:
is `token_delta` the bonds? 

in my head I like to group all positions into one of these units:
- base (underlying token, e.g. eth)
- shares (vault token, e.g. steth)
- bonds (aka hyBase; thing we mint to represent bond IOU)
- lp shares (thing we mint to represent participation in the pool's liquidity)
- withdrawal tokens (thing we mint to represent an IOU for receiving liquidity)

Reply:
You could alternatively do `base_or_share_delta` to make `as_base` relevant; and then leave `token_delta` to generically mean "everything besides base & shares", which I think is what you have now.

Reply:
Chatted about this offline, base delta is always base, vault share delta is always vault shares, and token delta is keyed by the `token_id`.